### ISSUE TITLE
A specific example in the documentation

### ISSUE THREAD

**Issue by etlabaume** [maxlaverse/terraform-provider-bitwarden#8]
I'd like to suggest adding an example on how to access custom fields stored in Bitwarden logins. It seems to be possible, however the syntax is unclear, at least to me.

**Comment by maxlaverse** [comment]
Hi @etlabaume !
After reading this [documentation section](https://www.terraform.io/language/functions/index_function) and seeing [this example](https://stackoverflow.com/questions/52119400/how-to-get-an-object-from-a-list-of-objects-in-terraform) (doesn't immediately answers your question), I found two ways to achieve what you want:

```terraform
element(
  data.bitwarden_item_login.<your-login-item>.field,
  index(data.bitwarden_item_login.<your-login-item>.field.*.name,"<name-of-the-field-your-looking-for>")
)
```

Or:
```terraform
lookup(
   zipmap(
     data.bitwarden_item_login.<your-login-item>field.*.name,
     data.bitwarden_item_login.<your-login-item>.field.*
   ),
   "<name-of-the-field-your-looking-for>"
)
```

With the second form you can eventually append `.text`, `.hidden` or `.boolean` to `data.bitwarden_item_login.<your-login-item>.field.*` if you know the type of field you're looking for. In both cases, make sure to have unique field names to avoid any trouble. Bitwarden allows to store multiple fields with the name but it would be harder to retrieve them.

It's a bit unfortunate the `type` information doesn't appear in the state actually. I'll probably try to change that, and/or document one of two examples.

Out of curiosity, what kind of information do you store in the field or more importantly, how do you use it, where do you need it ?

**Comment by etlabaume** [comment]
Hey, thanks for the quick response!

I managed to retrieve the custom field value that I was after using `data.bitwarden_item_login.<my-login-item>.field[n].text`, with n being the custom field number (starting with 0). Using the `name` of the field like you demonstrated is clearly more robust, I'm just not yet familiar with that syntax. Thanks for the tips!

To answer your last question, it's an API key that I store in the custom field, and I used it to configure other Terraform providers. Say I want to use [Cloudflare's provider](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs), I would configure it using:

provider "cloudflare" {
  email   = data.bitwarden_item_login.cloudflare.username
  api_key = data.bitwarden_item_login.cloudflare.field[0].text
}
